homeassistant/components/kef/media_player.py

```diff
@@ -0,0 +1,236 @@
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MAX_VOLUME, default=DEFAULT_MAX_VOLUME): cv.small_float,
+        vol.Optional(CONF_VOLUME_STEP, default=DEFAULT_VOLUME_STEP): cv.small_float,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the KEF platform."""
+    if DATA_KEF not in hass.data:
+        hass.data[DATA_KEF] = {}
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    name = config.get(CONF_NAME)
+    maximum_volume = config.get(CONF_MAX_VOLUME)
+    volume_step = config.get(CONF_VOLUME_STEP)
+
+    _LOGGER.debug(
+        "Setting up %s with host: %s, port: %s, name: %s, sources: %s",
+        DATA_KEF,
+        host,
+        port,
+        name,
+        KEF_LS50_SOURCES,
+    )
+
+    media_player = KefMediaPlayer(
+        name,
+        host,
+        port,
+        maximum_volume=maximum_volume,
+        volume_step=volume_step,
+        sources=KEF_LS50_SOURCES,
+        ioloop=hass.loop,
+    )
+    unique_id = media_player.unique_id
+    if unique_id in hass.data[DATA_KEF]:
+        _LOGGER.debug("%s is already configured.", unique_id)
+    else:
+        hass.data[DATA_KEF][unique_id] = media_player
+        async_add_entities([media_player], update_before_add=True)
+
+
+class KefMediaPlayer(MediaPlayerDevice):
+    """Kef Player Object."""
+
+    def __init__(self, name, host, port, maximum_volume, volume_step, sources, ioloop):
+        """Initialize the media player."""
+        self._name = name
+        self._sources = sources
+        self._speaker = AsyncKefSpeaker(
+            host, port, volume_step, maximum_volume, ioloop=ioloop
+        )
+
+        self._state = STATE_UNKNOWN
+        self._muted = None
+        self._source = None
+        self._volume = None
+        self._is_online = None
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    async def async_update(self):
+        """Update latest state."""
+        _LOGGER.debug("Running async_update")
+        try:
+            self._is_online = await self._speaker.is_online()
+            if self._is_online:
+                self._muted = await self._speaker.is_muted()
+                self._volume = await self._speaker.get_volume()
+                self._source, is_on = await self._speaker.get_source_and_state()
+                self._state = STATE_ON if is_on else STATE_OFF
+            else:
+                self._muted = None
+                self._source = None
+                self._volume = None
+                self._state = STATE_OFF
+        except Exception as err:
```

Comment:
Using Exception is too broad, that is why the Pylint Full Check is failing, see https://dev.azure.com/home-assistant/Home%20Assistant/_build/results?buildId=17094&view=logs&j=8b66295a-05e1-52cf-5640-7a4f4e1975e2&t=6c4465fe-285a-5650-787c-2e3dbe1da192&l=13

Reply:
I just want to catch all exceptions as to not crash HA when something happens.

But maybe HA already handles the errors itself? What happens if I do not do this?

Reply:
I understand your concern, but you should only use the exact Exceptions that could occur.

Not so sure about that, but I would assume that HA does not handle them automatically so updating the state would fail. Furthermore, the error message would be added to the log file.

Reply:
OK I now only catch the exceptions that `aiokef` throws.